ksqldb-functional-tests/src/test/resources/query-validation-tests/fk-join.json

```diff
@@ -5,7 +5,7 @@
       "statements": [
         "CREATE TABLE left_table (id1 BIGINT PRIMARY KEY, f1 BIGINT) WITH (kafka_topic='left_topic', format='JSON');",
         "CREATE TABLE right_table (id2 BIGINT PRIMARY KEY, f2 BIGINT) WITH (kafka_topic='right_topic', format='JSON');",
-        "CREATE TABLE output AS SELECT id1, id2, f1, f2 FROM left_table JOIN right_table ON id1 = f2;"
+        "CREATE TABLE output AS SELECT id1, f1, f2 FROM left_table JOIN right_table ON id1 = f2;"
```

Comment:
Should we rewrite those tests as `SELECT *` instead? What is the recommended test pattern?

Btw: #7526 move those cases to a new file.

Reply:
Added explicit tests for `SELECT *`, including with different combinations of qualifiers and aliases. I think it's fine to have the default tests use explicit selects rather than `SELECT *` as long as both are covered.